colorbleed/plugins/maya/load/load_yeti_cache.py

```diff
@@ -83,30 +84,88 @@ def remove(self, container):
-
-        # TODO: Count the amount of nodes cached
-        # To ensure new nodes get created or old nodes get destroyed
-
-        for node in yeti_node:
-            # Remove local given namespace
-            node_name = node.split(namespace, 1)[-1]
-            file_name = node_name.replace(":", "_")
-
-            # Check if the node has a cache
-            tmp_cache = os.path.join(path, "{}.%04d.fur".format(file_name))
-            fpath = self.validate_cache(os.path.normpath(tmp_cache))
 
-            # Update the attribute
-            cmds.setAttr("{}.cacheFileName".format(node), fpath, type="string")
-
-        # Update the container
-        cmds.setAttr("{}.representation".format(container["objectName"]),
+        # Get all node data
+        fname, ext = os.path.splitext(path)
+        settings_fname = "{}.fursettings".format(fname)
+        with open(settings_fname, "r") as fp:
+            settings = json.load(fp)
+
+        # Collect scene information of asset
+        set_members = cmds.sets(container["objectName"], query=True)
+        container_root = lib.get_container_transforms(container,
+                                                      members=set_members,
+                                                      root=True)
+        scene_nodes = cmds.ls(set_members, type="pgYetiMaya", long=True)
+
+        # Build lookup with cbId as keys
+        scene_lookup = defaultdict(list)
+        for node in scene_nodes:
+            cb_id = lib.get_id(node)
+            scene_lookup[cb_id].append(node)
+
+        # Re-assemble metadata with cbId as keys
+        meta_data_lookup = {n["cbId"]: n for n in settings["nodes"]}
+
+        # Compare look ups and get the nodes which ar not relevant any more
+        to_delete_lookup = {cb_id for cb_id in scene_lookup.keys() if
+                            cb_id not in meta_data_lookup}
+        if to_delete_lookup:
+
+            # Get nodes and remove entry from lookup
+            to_remove = []
+            for _id in to_delete_lookup:
+                # Get all related nodes
+                shapes = scene_lookup[_id]
+                # Get the parents of all shapes under the ID
+                transforms = cmds.listRelatives(shapes, parent=True,
+                                                fullPath=True) or []
+                to_remove.extend(shapes + transforms)
+
+                # Remove id from look uop
+                scene_lookup.pop(_id, None)
+
+            cmds.delete(to_remove)
+
+        for cb_id, data in meta_data_lookup.items():
+
+            # Update cache file name
+            file_name = data["name"].replace(":", "_")
+            cache_file_path = "{}.%04d.fur".format(file_name)
+            data["attrs"]["cacheFileName"] = os.path.join(path, cache_file_path)
+
+            if cb_id not in scene_lookup:
+                new_nodes = self.create_nodes(namespace, [data])
+                cmds.sets(*new_nodes, addElement=container_node)
```

Comment:
Doesn't this also work without the `*`? Like `cmds.sets(new_nodes, addElement=container_node)`

Reply:
Yes it should, don't know why I used this one :thinking: